### PR TITLE
Add mocks to test ProtocolManager

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- To test ProtocolManager, mocks are added.
- `consensus.Engine`, `cn.txPool` and `cn.blockChain` are the ones mocked.
- Added more functions to `cn.blockChain` interface to change `blockchain.BlockChain` from `NewProtocolManager`.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ehnuje** [comment]
@jeonghonah @nogang @KimKyungup @andybclee Dear reviewers, could you please review this change?

**Comment by ehnuje** [comment]
@KimKyungup I wanted to leave the private interfaces as-is, however, it is better to follow your suggestion. Let me update the change. Thanks!

File: node/cn/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		header.Hash()
+		block := types.NewBlockWithHeader(header)
+
+		mockBlockChain.EXPECT().CurrentBlock().Return(block).Times(2)
+		mockEngine.EXPECT().Protocol().Return(consensus.Protocol{}).Times(1)
+
+		pm, err := NewProtocolManager(nil, downloader.FastSync, 0, nil, mockTxPool,
+			mockEngine, mockBlockChain, nil, -1, &Config{})
+
+		assert.Nil(t, pm)
```
Comment (jeonghonah) [inline node/cn/handler_test.go]: Is there any reason to test failing case only?

File: node/cn/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/klaytn/klaytn/consensus"
+	mocks2 "github.com/klaytn/klaytn/consensus/mocks"
+	"github.com/klaytn/klaytn/datasync/downloader"
+	"github.com/klaytn/klaytn/node/cn/mocks"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+var num1 = uint64(20190902)
```
Comment (jeonghonah) [inline node/cn/handler_test.go]:
```suggestion
var blkNum1 = uint64(20190902)
```

File: node/cn/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	return mockEngine, mockBlockChain, mockTxPool
+}
+
+func TestNewProtocolManager(t *testing.T) {
+	//1. If consensus.Engine returns an empty Protocol, NewProtocolManager throws an error.
+	{
+		mockEngine, mockBlockChain, mockTxPool := newMocks(t)
+
+		header := &types.Header{Number: big.NewInt(int64(num1))}
```
Comment (jeonghonah) [inline node/cn/handler_test.go]:
```suggestion
		header := &types.Header{Number: big.NewInt(int64(blkNum1))}
```

File: node/cn/handler_test.go (hunk 1 above)
Comment (ehnuje) [reply node/cn/handler_test.go]: @jeonghonah To call NewProtocolManager successfully, there are many structs to be handled properly. Not to make the change large and just introduce the mock usage, I only made a failing test case.

**Pull request merged**